### ISSUE TITLE
How can I get an index created before to continue assinging doctytpes to the created index?

### ISSUE THREAD

**Issue by juandasgandaras** [elastic/elasticsearch-dsl-py#280]
To create an index and associate a doctype we use this:

```
# create the index object
index = Index('my_index')
# associate the document types with it
index.doc_type(my_doc_type0)
# create it in elasticsearch
index.create()
```

But my question is how can I get an index created before to continue assinging doctytpes to the created index.

**Comment by HonzaKral** [comment]
I am afraid I don't understand what you'd like to do. There is nothing stopping you from just creating the index object in python, regardless of whether it exists in elasticsearch or not. Or do you wish to pull it's definition (including the mappings) from es?

Thanks!

**Comment by juandasgandaras** [comment]
The thing that I want is to check before if this index exist and if this is not this case to create and to do the mapping.

Finally I resolved but with the Elasticsearch python library:
```
    es = Elasticsearch()
    exist = es.indices.exists(my_index)
    
    if not(exist):
        # create the index object
        index = Index(my_index)

        # associate the document types with it
        index.doc_type(my_doc_type)
        index.doc_type(my_other_doc_type)
            
        # create it in elasticsearch
        index.create()
```

and after in the injection with the doc_type class I do:

```
e = my_doc_type(_index=my_index, another_parameters ...)

```

**Comment by HonzaKral** [comment]
Well, everything until the `index.create()` you should be doing no matter whether the index already exists in Elasticsearch.

Would a `.exists()` method on the `Index` class help? Or do you think it would be better have the `create` method accept something like `ignore_if_exists`?

**Comment by juandasgandaras** [comment]
In my opinion the two solutions would be well, I think. 

Because sometimes it would be necesary to check if this index exists or not so the other option too.

Anyway your library is so useful and to ease my job.

Thanks for your job and your help.

**Comment by HonzaKral** [comment]
I implemented `exists` in ad650675fe883572922dcbb18e959019a159e0a9 Note that you can always say `ignore=400` when calling `create` which will ignore (not throw exception) 400 responses, such as when an index already exists.

**Comment by juandasgandaras** [comment]
Thanks for your job
